rustls-platform-verifier/src/verification/mod.rs

```diff
@@ -77,3 +77,29 @@ fn invalid_certificate(reason: impl Into<String>) -> rustls::Error {
 /// - id-kp-serverAuth
 // TODO: Chromium also allows for `OID_ANY_EKU` on Android.
 pub const ALLOWED_EKUS: &[&str] = &["1.3.6.1.5.5.7.3.1"];
+
+#[cfg(any(target_os = "macos", target_os = "ios", target_os = "android", windows))]
+impl Verifier {
```

Comment:
Is it possible to extend support for supplying the crypto provider to unix/linux as well?

I think it's largely a matter of wiring up `WebPkiServerVerifier::builder_with_provider()` where we're using `WebPkiServerVerifier::builder()`: https://github.com/rustls/rustls-platform-verifier/blob/b6f6334dc9d56e53d4a722ab801d76b935250e04/rustls-platform-verifier/src/verification/others.rs#L88-L90